collector: Implement uname collector for FreeBSD

phyber opened the pull request:
This PR implements the uname collector for FreeBSD.

  - Breaks out the common part into `uname.go`
  - Adds a `uname` type which is a `struct` of strings named after their `unix.Utsname` counterparts.
    - OS specific methods return the `uname` for consumption by the collector in `uname.go`
  - Leaves the Linux collector in `uname_linux.go`
  - Adds a FreeBSD collector in `uname_freebsd.go`

FreeBSD does not have a `uname.Domainname` like Linux, so we attempt to emulate it in order to enable comparing `node_uname_info` across operating systems. Perhaps a user wants to count all nodes under a specific domainname, etc.

In the event a domain name cannot be split out of the `uname.Nodename`, we set the domainname to `(none)`, which appears to be what happens on Linux when uname doesn't know the domainname for some reason.

I'm not particularly attached to this behaviour, and would be happy to drop the `domainname` label from the FreeBSD uname info if the maintainers preferred.

Tagging @SuperQ and @discordianfish as per contribution guidelines.

SuperQ commented on collector/uname.go:
```diff
@@ -0,0 +1,45 @@
+// Copyright 2015 The Prometheus Authors
```
This should be 2019, as it's the date of the new file.

phyber commented:
Thanks! 👍

phyber commented:
The issue appears to be fixed now, and was that I was building `uname.go` for all OSs, even those without `uname_${OS}.go` files. This is now fixed, and I've moved the `Update` method back into `uname.go`, with a callout to a `getUname()` function.

This means the uname collector now follows the style of how the same type of thing is handled in other collectors.

SuperQ commented on collector/uname.go:
```diff
@@ -0,0 +1,65 @@
+)
+
+type unameCollector struct{}
+type unameOutput map[string]string
```
I would prefer to make this a struct.
```
type uname struct {
  SysName string,
  Release string,
...
}

phyber replied:
Thanks, that makes much more sense. Not sure what I was thinking.

phyber commented:
No problem, on it.

SuperQ commented:
Thanks, will merge when the tests go green. :tada:
